contracts/strategies/RFPStrategy.sol

```diff
@@ -0,0 +1,278 @@
+
+    /// ===============================
+    /// ========== Events =============
+    /// ===============================
+
+    event MAX_BID_UPDATED(uint256 maxBid);
+    event MILESTONE_SUBMITTED(uint256 milestoneId);
+    event MILESTONE_REJECTED(uint256 milestoneId);
+    event MILESTONES_SET();
+
+    /// ================================
+    /// ========== Storage =============
+    /// ================================
+
+    bool public registryGating;
+    uint256 public maxBid;
+    uint256 public upcomingMilestone;
+    address public acceptedRecipientId;
+
+    address[] private _recipientIds;
+    Milestone[] public milestones;
+
+    /// @notice recipientId -> Recipient
+    mapping(address => Recipient) public recipients;
+
+    /// ===============================
+    /// ======== Constructor ==========
+    /// ===============================
+
+    constructor(address _allo, string memory _name) BaseStrategy(_allo, _name) {}
+
+    /// ===============================
+    /// ========= Initialize ==========
+    /// ===============================
+
+    function initialize(uint256 _poolId, bytes memory _data) public override {
+        super.initialize(_poolId, _data);
+
+        (maxBid, registryGating) = abi.decode(_data, (uint256, bool));
+
+        emit MAX_BID_UPDATED(maxBid);
+    }
+
+    /// ===============================
+    /// ============ Views ============
+    /// ===============================
+
+    /// @notice Checks if msg.sender is eligible for RFP allocation
+    /// @param _recipientId Id of the recipient
+    function getRecipientStatus(address _recipientId) public view override returns (RecipientStatus status) {
+        return recipients[_recipientId].recipientStatus;
+    }
+
+    /// @notice Returns the payout summary for the accepted recipient
+    function getPayouts(address[] memory, bytes memory, address) external view returns (PayoutSummary[] memory) {
+        PayoutSummary[] memory payouts = new PayoutSummary[](1);
+        payouts[0] = PayoutSummary(acceptedRecipientId, recipients[acceptedRecipientId].proposalBid);
+
+        return payouts;
+    }
+
+    /// @notice Checks if address is elgible allocator
+    /// @param _allocator Address of the allocator
+    function isValidAllocator(address _allocator) public view returns (bool) {
+        return allo.isPoolManager(poolId, _allocator);
+    }
+
+    /// @notice Get the status of the milestone
+    /// @param _milestoneId Id of the milestone
+    function getMilestoneStatus(uint256 _milestoneId) public view returns (RecipientStatus) {
+        return milestones[_milestoneId].milestoneStatus;
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Set milestones for RFP pool
+    /// @param _milestones The milestones to be set
+    function setMilestones(Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
```

Comment:
function allows overwriting old and paid milestones
I'm not sure if it's a problem. just leave this note here.

wdyt about:
```solidity
        uint256 paidMilestones = upcomingMilestone == 0 ? 0 : upcomingMilestone - 1;
        uint256 milestonesLength = _milestones.length;
        if(milestoneLength <= paidMilestones) {
             revert INVALID_MILESTONE();
        } else {
             milestoneLength -= paidMilestones;
        }
        for (uint256 i = paidMilestones; i < milestonesLength;) {
            milestones.push(_milestones[i]);

            unchecked {
                i++;
            }
        }
```

Reply:
Ah i see what you're saying but here is what i think what we should update it to 
```
        if (upcomingMilestone != 0) {
            revert MILESTONES_ALREADY_SET();
        }
```

Are you thinking we should allow partial setting of milestones ?